src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java

```diff
@@ -397,6 +401,22 @@ public TagSet queryMetricTags(DatastoreMetricQuery query) throws DatastoreExcept
 		return (tagSet);
 	}
 
+	@Override
+	public void indexMetricTags(DatastoreMetricQuery query) throws DatastoreException
+	{
+		CQLBatch batch = m_cqlBatchFactory.create();
+		Iterator<DataPointsRowKey> rowKeys = getKeysForQueryIterator(query);
+
+		MemoryMonitor mm = new MemoryMonitor(20);
+		while (rowKeys.hasNext())
+		{
+			DataPointsRowKey dataPointsRowKey = rowKeys.next();
+			batch.indexRowKey(query.getName(), dataPointsRowKey, 0);
```

Comment:
This creates far too many insert statements. It seems like it's creating an insert statement for every single datapoint, which I assume many are duplicates. It needs to either dedup insert statements or combine them somehow to be performant.

Reply:
This looks fine to me.  It will generate a set of inserts for every row key.

Reply:
we do need to get the ttl from the row key and pass it back in to preserve it.

Reply:
Awesome. I was having issues because I was reinserting all the data back in. After making modifications based on your comments below this works out a lot better.

Reply:
Any tips on getting the TTL from the row key? I don't see a getter.